### ISSUE TITLE
Last task in workflow doesn't update to succeeded in graph GUI

### ISSUE THREAD

**Issue by jfrost-mo** [cylc/cylc-ui#1943]
### Description

After a workflow reaches its final cycle point, and the last task completes, the reported state in the graph view of the GUI still claims the task to be running, when it is not.

### Reproducible Example

Run a workflow, opening it in cylc gui on the graph view. Once the workflow reaches its final cycle point, and the last task completes, observe the reported state of that task on the graph. It will be left in a running state, despite actually having succeeded.

### Expected Behaviour

The task in the GUI should be marked as succeeded.

**Comment by Fraetor** [comment]
An example. Note the workflow is stopped after having completed, but the task still claims to be running.

![20240924_122816](https://github.com/user-attachments/assets/d6ccb7f5-a8c3-4d6c-b70b-b0ed9f0f94a7)

**Comment by MetRonnie** [comment]
Duplicate of #1756

**Comment by oliver-sanders** [comment]
I think that's a different issue.

@jfrost-mo, if you switch back to the tree view, the task should still be visible there too, what state is it in?

**Comment by MetRonnie** [comment]
I can reproduce with a simple workflow. 

> if you switch back to the tree view, the task should still be visible there too, what state is it in?

The workflow completed so the tree view is empty after shutdown

**Comment by oliver-sanders** [comment]
Good point, would need an incomplete task to test the graph view against the tree view.

**Comment by MetRonnie** [comment]
Ok so not a duplicate because #1756 is just about clearing the graph view but the bug here is we are missing a delta update for the final state of the task before it is pruned
